WordPress/Tests/WP/CapabilitiesUnitTest.inc

```diff
@@ -0,0 +1,88 @@
+add_management_page( 'page_title', 'menu_title', 'editor', 'menu_slug', 'function' ); // Error.
+add_options_page( 'page_title', 'menu_title', 'level_10', 'menu_slug', 'function' ); // Error.
+if ( current_user_can( 'super_admin' ) ) { } // Error.
+if( current_user_can_for_blog( '1', 'editor' ) ) { }  // Error.
+add_dashboard_page(
+	'page_title',
+	'menu_title',
+	'super_admin' /* Error */, // Error.
+	'menu_slug',
+	'function'
+);
+add_utility_page(
+	'page_title'
+	,'menu_title'
+	,'super_admin' // Error.
+	,'menu_slug'
+	,'function'
+	,'icon_url'
+);
+
+
+add_menu_page( $pagetitle, $menu_title, $subscriber, 'handle', 'function', 'icon_url' ); // Warning.
+add_plugins_page( 'page_title', 'menu_title', $cap, 'menu_slug', 'function' ); // Warning.
+add_users_page( 'page_title', 'menu_title', 'foo_bar', 'menu_slug', 'function' ); // Warning.
+add_management_page( 'page_title', 'menu_title', 'foo_bar', 'menu_slug', 'function' ); // Warning.
+add_options_page( $pagetitle, $menu_title, CONSTANT, 'menu_slug', 'function' ); // Warning.
+add_menu_page( $pagetitle, 'menu_title', 'foo_bar', 'handle', 'function', 'icon_url' ); // Warning.
+if ( author_can( $post, $capability ) ) { } // Warning.
+add_submenu_page(
+	'parent_slug',
+	'page_title',
+	'menu_title',
+	$variable, // Warning.
+	'menu_slug',
+	'function'
+);
+if ( author_can( $post, 'custom_cap' ) ) { } // Warning.
+
+// phpcs:set WordPress.WP.Capabilities custom_capabilities[] custom_cap,foo_bar
+if ( current_user_can( 'foo_bar' ) ) { } // OK.
+if ( author_can( $post, 'custom_cap' ) ) { } // OK.
+
+// phpcs:set WordPress.WP.Capabilities custom_capabilities[]
+
+if ( current_user_can( 'foo_bar' ) ) { } // Warning.
+if ( current_user_can_for_blog( '3', 'custom_cap' ) ) { } // Warning.
+
+
+
+if ( author_can( $post, 'read' ) ) { } // OK.
+if ( author_can( $post, '' ) ) { } // Error.
+if ( author_can( $post, 'custom_cap' ) ) { } // Warning.
+
+map_meta_cap( 'edit_posts', $user->ID ); // OK.
+map_meta_cap( 'editor', $user->ID ); // Error.
+
+add_menu_page(
+	$p,
+	$t, // Comment.
+	$capability, // Low severity warning.
+);
+
+add_menu_page( $p, $t, 'admin' . 'istrator' ); // Low severity warning.
+
+// PHP 8.0 named parameters support.
+add_menu_page( capability: 'foobar', page_title: $p, menu_title: $m ); // Warning.
+
+add_menu_page($p, $t, $caps['level'] ); // Low severity warning.
+
+add_menu_page($p, $t, 'level_' . $level ); // Low severity warning.
+
+// Parse error, but just making sure we account for all possibilities.
+add_menu_page($p, $t, 'level_' 'level' ); // Low severity warning.
+
+// Another parse error, but the sniff should still handle this correctly (by bowing out).
+add_menu_page( $p, $t, /* deliberately empty */, $slug, );
+
+add_posts_page( 'page_title', 'menu_title', self /* comment */   ::   CAPABILITY, 'menu_slug', 'function' ); // Warning.
+add_posts_page( 'page_title', 'menu_title', 'admin' /* comment */   . 'istrator', 'menu_slug', 'function' ); // Warning.
```

Comment:
Noticed some things we don't have tests for. It's arbitrary whether or not to add them as they don't necessarily test the functionality of the sniff itself, but rather the functionality of the abstract sniff, but then again, the abstract sniff doesn't have its own tests, so until then, it kind of makes sense to have some tests for these...:
* Parse error/live coding, i.e. `add_posts_page( 'page_title',` (and nothing else - should be last test in the test file).
* `$obj->current_user_can()`
* `My\NS\add_posts_page(...)`
* etc